mitodl/open-discussions
# Correct issues with documentation for offered_by filter on courses endpoint

#### Pre-Flight checklist


#### What are the relevant tickets?
closes https://github.com/mitodl/open-discussions/issues/3922

#### What's this PR do?
This PR fixes the erroneously marked `offered_by` endpoint in the `drf-spectacular` docs so that it shows the correct type and works in the "try it now" section.

This is done by using `extend_schema_field` which removes the other documentation, so I have put them back to be consistent with other filters, such as audience.

#### How should this be manually tested?
Go to `http://od.odl.local:8063/api/v0/schema/swagger-ui/#/courses/courses_list` and you should see that the type is `string` and the enum & choices should be correct and match the[ OfferedBy constant](https://github.com/mitodl/open-discussions/blob/51909e81d663e9ef6f6c517f2176c90b12704b6f/course_catalog/constants.py#L7).

Click the `Try it now` button and you should be able to filter by `offeredby` successfully now.
